include/unifex/let_value_with_stop_token.hpp

```diff
@@ -0,0 +1,234 @@
+/*
+ * Copyright (c) Facebook, Inc. and its affiliates.
+ *
+ * Licensed under the Apache License Version 2.0 with LLVM Exceptions
+ * (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ *   https://llvm.org/LICENSE.txt
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+#pragma once
+
+#include <unifex/config.hpp>
+#include <unifex/execution_policy.hpp>
+#include <unifex/inplace_stop_token.hpp>
+#include <unifex/just.hpp>
+#include <unifex/let_value.hpp>
+#include <unifex/receiver_concepts.hpp>
+#include <unifex/sender_concepts.hpp>
+#include <unifex/type_list.hpp>
+
+#include <unifex/detail/prologue.hpp>
+
+namespace unifex {
+
+namespace _let_v_w_stop_tok {
+
+template <typename Operation, typename Receiver>
+struct _stop_token_receiver {
+  class type;
+};
+
+template <typename InnerOp, typename Receiver>
+struct _stop_token_operation {
+  struct type;
+};
+
+template <typename InnerOp, typename Receiver>
+using operation = typename _stop_token_operation<InnerOp, Receiver>::type;
+
+template <typename Operation, typename Receiver>
+using stop_token_receiver =
+    typename _stop_token_receiver<Operation, Receiver>::type;
+
+template <typename Operation, typename Receiver>
+class _stop_token_receiver<Operation, Receiver>::type {
+public:
+  explicit type(inplace_stop_token stop_token, Receiver&& r) noexcept(
+      unifex::is_nothrow_move_constructible_v<Receiver>)
+    : stop_token_(stop_token)
+    , receiver_{(Receiver &&) r} {}
+
+  template(typename... Values)(requires receiver_of<Receiver, Values...>) void set_value(
+      Values&&... values) noexcept(is_nothrow_receiver_of_v<Receiver, Values...>) {
+    unifex::set_value(std::move(receiver_), (Values &&) values...);
+  }
+
+  template(typename Error)(requires receiver<Receiver, Error>) void set_error(
+      Error&& error) noexcept {
+    unifex::set_error(std::move(receiver_), (Error &&) error);
+  }
+
+  void set_done() noexcept { unifex::set_done(std::move(receiver_)); }
+
+  friend inplace_stop_token
+  tag_invoke(tag_t<get_stop_token>, const type& r) noexcept {
+    return r.stop_token_;
+  }
+
+  template(typename CPO, typename Self)(
+      requires is_receiver_query_cpo_v<CPO> AND same_as<
+          Self,
+          type>) friend auto tag_invoke(CPO cpo, const Self& self) noexcept(is_nothrow_callable_v<CPO, const Receiver&>)
+      -> callable_result_t<CPO, const Receiver&> {
+    return cpo(self.receiver_);
+  }
```

Comment:
The formatting here is what Eric meant when he said Clang Format "makes a hash" of the concept-alike macros.
```suggestion
  template(typename... Values)
      (requires receiver_of<Receiver, Values...>)
  void set_value(Values&&... values) noexcept(is_nothrow_receiver_of_v<Receiver, Values...>) {
    unifex::set_value(std::move(receiver_), (Values &&) values...);
  }

  template(typename Error)
      (requires receiver<Receiver, Error>)
  void set_error(Error&& error) noexcept {
    unifex::set_error(std::move(receiver_), (Error &&) error);
  }

  void set_done() noexcept { unifex::set_done(std::move(receiver_)); }

  friend inplace_stop_token
  tag_invoke(tag_t<get_stop_token>, const type& r) noexcept {
    return r.stop_token_;
  }

  template(typename CPO, typename Self)
      (requires is_receiver_query_cpo_v<CPO> AND same_as<Self, type>)
  friend auto tag_invoke(CPO cpo, const Self& self) noexcept(is_nothrow_callable_v<CPO, const Receiver&>)
      -> callable_result_t<CPO, const Receiver&> {
    return cpo(self.receiver_);
  }
```

I have had success "tricking" Clang Format by `#if 0`ing out the `template` and `requires` clauses, running Clang Format, and then manually formatting the `template` and `requires` clauses afterwards.